### ISSUE TITLE
Automatically passing the file and lines

### ISSUE THREAD

**Issue by LecrisUT** [aradi/fortuno.old#2]
From what I understand in the current implementation, e.g.:
https://github.com/aradi/fortuno/blob/c5a08e62c6c06bb95e54b233be46f696854ed874/src/fortuno/genericcontext.f90#L32-L51
The user has to pass the file and line manually to create the failure message.  Wouldn't it be appropriate to define a macro to pass `__FILE__` and `__LINE__` to automatically pass these informations?

**Comment by aradi** [comment]
Absolutely. The plan is to have a solid framework first without the need for any preprocessors. Then I want to add an include file for  cpp/fpp and fypp, respectively,  containing appropriate wrappers for `check()`, which pass the necessary information automatically through the call.
